src/api/getArticle.js

```diff
@@ -1,7 +1,14 @@
 import { cachedFetch, buildPcsUrl, canonicalizeTitle } from 'utils'
 
-export const getArticle = (lang, title) => {
+export const getArticle = (lang, title, i18n) => {
   const url = buildPcsUrl(lang, title, 'mobile-sections')
+
+  // retrieve content language and set it back to app language
```

Comment:
I wonder how often we are going to have to do this `locale` quick switch to grab a label, perhaps a good candidate for a `utils` helper in the future?

Reply:
yeah, I like the idea, banana doesn't support this, we can do our own version of getting translation once in the other language

Reply:
Currently, `useI18n()` can also be used in the component level, we need a way to make `banana` available in utils as well, let's do it in another refactor pr including the change for search language and article footer.

Reply:
I can create Phabricator ticket

Reply:
https://phabricator.wikimedia.org/T248354

Reply:
The banana instance is a stateful and expensive object so it is made available through a context. That makes it unsuitable for utils. On the other hand I agree that the way we use it with the content language should be refactored. We could do something like `const i18nInContentLang = useI18n(article.contentLang)`

Building the UI with localized strings should stay in components. In this case, user-created article content needs to be modified and cached in the api's transform function. I would prefer that `useArticle` passes only the needed string(s) to `getArticle` instead of the banana instance.

Reply:
Done passing the translation instead of i18n here bdfe89b4ccc782aebffc7410ec413d4c06342f80